corehq/messaging/scheduling/forms.py

```diff
@@ -209,15 +209,22 @@ def set_app_and_form_unique_id_choices(self):
         self.fields['app_and_form_unique_id'].choices = [('', '')] + self.schedule_form.form_choices
 
     def clean_subject(self):
-        if self.schedule_form.cleaned_data.get('content') not in (ScheduleForm.CONTENT_EMAIL,
-                                                                  ScheduleForm.CONTENT_FCM_NOTIFICATION):
+        if (self.schedule_form.cleaned_data.get('content') == ScheduleForm.CONTENT_FCM_NOTIFICATION
+                and self.cleaned_data['fcm_message_type'] == FCMNotificationContent.MESSAGE_TYPE_NOTIFICATION):
```

Comment:
This field `subject` is available only for `Notification` message type and not for `Data` message type hence the check.

One advantage of doing at field level instead at `clean` is the error on the UI appears just below the field ; there is a catch though- if we are using any other field like `fcm_message_type` in this case, it should be before the field being validated.
I add max length on `subject` and `message` field for FCM content type in later [commit](https://github.com/dimagi/commcare-hq/pull/33047/commits/2dab3eea2b912258cd8e3b58f32fe081066a628f)

Reply:
> This field subject is available only for Notification message type and not for Data message type hence the check.

Not sure how that makes a difference since the same if being used here will be used in clean method.

> One advantage of doing at field level instead at clean is the error on the UI appears just below the field

That is a good point. I believe you could set an error for a specific field in 'clean' as well or is that not the right thing to do?

Reply:
> Not sure how that makes a difference since the same if being used here will be used in clean method.

Right, I was just communicating the reason for the check. 

> That is a good point. I believe you could set an error for a specific field in 'clean' as well or is that not the right thing to do?

Right (Noting this) -  we should be able set an error for a specific field in `clean` method as well. 

On rethinking this , I think your suggestion of having this on clean method is more intuitive and consistent.
> I have not worked with forms a lot, but I think if we are to check for more than one field together, it would go in the clean method. Good to confirm.